### PR TITLE
Added async call to Google sheets API

### PR DESCRIPTION
**Pull request by Alfradu**
The syncronous calls to google sheet's api made ui unresponsive until http timeout is reached or data is returned. This update asyncronously pulls data from google - however slow the api might be at a time and leaves the RL client operational throughout updating the plugin with data.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,7 +10,7 @@ repositories {
 	mavenCentral()
 }
 
-def runeLiteVersion = '1.8.24.2'
+def runeLiteVersion = '+'
```
Comment (Alfradu) [inline build.gradle:13]: Keeping Runelite up to latest version so we can test plugins.

File: src/main/java/com/clanevents/GoogleSheet.java [hunk 2]
```diff
@@ -55,18 +55,16 @@ private static Sheets getSheets() {
         NetHttpTransport transport = new NetHttpTransport.Builder().build();
         JacksonFactory jsonFactory = JacksonFactory.getDefaultInstance();
         HttpRequestInitializer httpRequestInitializer = request -> {
-            request.setReadTimeout(httpTimeout * 1000);
+            //request.setReadTimeout(httpTimeout * 1000);
```
Comment (Alfradu) [inline src/main/java/com/clanevents/GoogleSheet.java:58]: dont need a timeout if we are async fetching data.

**Comment by brianmm1994** [comment]
That sounds like a pretty good fix, I know that was the main issue I had when making all of my updates a while back. For some reason Google Docs would just randomly become unresponsive and you'd have to wait until the HTTP timeout finishes. I had resolved it by adding a plugin setting that allows users to modify the HTTP timeout value, and I gave it a default value of like 1s or something, but I know it wasn't a very good solution. I'm assuming you've done a decent amount of testing already? I'll see if I can get cmsu224 to test it out and merge it if everything goes well.

**Comment by Brianmm94** [comment]
That sounds like a pretty good fix, I know that was the main issue I had when making all of my updates a while back. For some reason Google Docs would just randomly become unresponsive and you'd have to wait until the HTTP timeout finishes. I had resolved it by adding a plugin setting that allows users to modify the HTTP timeout value, and I gave it a default value of like 1s or something, but I know it wasn't a very good solution. I'm assuming you've done a decent amount of testing already? I'll see if I can get cmsu224 to test it out and merge it if everything goes well. I might test it out a bit as well if I can find the time.

**Comment by Alfradu** [comment]
Yes! testing has been done, would be great to have cmsu224 to test as well to waterproof the changes.

Happy to hear that you like the changes. If you want I could write out a more descriptive changelog of what we have messed around with.

File: src/main/java/com/clanevents/ClanEventsPanel.java [hunk 3]
```diff
@@ -201,6 +187,34 @@ public void mousePressed(MouseEvent e) {
...
         //Add the plugin's main panel
         this.add(ssArea, BorderLayout.NORTH);
+        service.addListener(this);
+        if (service.state.equals(State.IDLE)) service.refreshData();
+    }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent evt) {
+        if (evt.getPropertyName().equals("state") && evt.getNewValue() != evt.getOldValue()) {
+            System.out.printf("State changing: old: %s, new: %s%n", evt.getOldValue(), evt.getNewValue());
```
Comment (YvesW) [inline src/main/java/com/clanevents/ClanEventsPanel.java]:
```suggestion
            //System.out.printf("State changing: old: %s, new: %s%n", evt.getOldValue(), evt.getNewValue());
```
It's a bit annoying when testing your own plugins to have your console spammed with println or printf from installed plugins. Would be lovely if you could comment it out!

Comment (Alfradu) [reply src/main/java/com/clanevents/ClanEventsPanel.java]: removed prints

**Pull request merged**